### ISSUE TITLE
Questions about repo

### ISSUE THREAD

**Issue by adrianerrea** [XiongjunGuan/JIPNet#4]
Hi there,

First of all, thanks for sharing the inference code and the resources here. Great job!

I got several questions referring to this repo:

1. Are you planning to share the training process? Code, data used, validation and so on. It would be really useful for other developers
2. About pytorch models you shared in the repo, are they available to use them with MIT license as same as JIPNet?
3. From the JIPNet.py file, the forward function is the following:

        def forward(self, inps):

                batch_dim = inps[0].shape[0]

                x = torch.cat(inps, dim=0)

                x = self.encoder(x)

                x = self.mid_layer(x)

                [x0, x1] = torch.split(x, [batch_dim, batch_dim], dim=0)

                cla_pred, align_pred = self.transformer_layer(x0, x1)

                cla_pred = torch.sigmoid(cla_pred)

                return cla_pred, align_pred

If we want to get just the feature of the input, I guess we can reduce it to:

         def extract_features(self, inps):

                batch_dim = inps[0].shape[0]

                x = torch.cat(inps, dim=0)

                x = self.encoder(x)

                return self.mid_layer(x)

Am I right? If so, the feature has a shape of ([1, 264, 20, 20]). Can we flatten that matrix into a 1D array and compare it against other flattened 1D array?

Thanks in advance!

**Comment by XiongjunGuan** [comment]
Thank you for your recognition! About your questions:

1.	About Sharing the Training Process:
Currently, we have no plans to release the full training code, data, and validation pipeline. However, once we complete some of the ongoing work, we will consider organizing and sharing this content.

2.	About License:
Yes, the PyTorch models shared in this repository are all available under the MIT license, just like JIPNet. 

3.	Feature Extraction:
Yes, your approach of extracting features is effective. Also, if you are interested in obtaining features that are less affected by fingerprint modality, you may find the intermediate features of my other repo (about fingerprint enhancement) helpful. https://github.com/XiongjunGuan/FpEnhancer 

If you have any questions, please feel free to contact with me.

**Comment by adrianerrea** [comment]
Awesome! Thanks!
